Reflection_Engine/Query/Description.cs

```diff
@@ -54,23 +54,50 @@ public static string Description(this ParameterInfo parameter)
         {
             IEnumerable<InputAttribute> inputDesc = parameter.Member.GetCustomAttributes<InputAttribute>().Where(x => x.Name == parameter.Name);
             if (inputDesc.Count() > 0)
+            {
                 return inputDesc.First().Description;
-            else if (parameter.ParameterType != null)
-                return parameter.Name + " is a " + parameter.ParameterType.ToText();
-            else
+            }
+            if (parameter.ParameterType == null)
+            {
                 return "";
+            }
+            return parameter.ParameterType.Description();
         }
 
         /***************************************************/
 
         [Description("Return the custom description of a C# class")]
         public static string Description(this Type type)
         {
+            if (type == null)
+            {
+                return "";
+            }
+
             DescriptionAttribute attribute = type.GetCustomAttribute<DescriptionAttribute>();
             if (attribute != null)
+            {
                 return attribute.Description;
-            else
-                return "This is a " + type.ToText();
+            }
+            else if (type.IsInterface)
+            {
+                string echo = $"This is a {type.ToText()}: ";
+
+                List<Type> t = type.ImplementingTypes();
+                int m = Math.Min(15, t.Count);
+
+                for (int i = 0; i < m; i++)
```

Comment:
Little trick here, you can use Linq to save yourself the trouble of dealing with the final ',':

`t.Select(x => x.ToText()).Aggregate((a,b) => a + ", " + b)`

On a side note while I am here, try to avoid using one-letter variable names like 't' for other things than loops (or Linq  methods)